### PR TITLE
rfb: add RFB frames

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://suricata.readthedocs.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/5717
 
Previous PR: #8312 

Describe changes:
- Fix rust check

suricata-verify-pr: 1045

### CODE REVIEW

**Comment by hsadia538** [comment]
I have written a suricat-verify test for the Pdu frame and it's working, Should I make a suricata-verify pr as well along with this?

**Comment by victorjulien** [comment]
> I have written a suricat-verify test for the Pdu frame and it's working, Should I make a suricata-verify pr as well along with this?

Yes, please make it first and reference it in your suricata PR by adding a line in the message `suricata-verify-pr: NNN` where `NNN` is the SV PR number.

File: rust/src/rfb/rfb.rs [hunk 1]
```diff
@@ -151,8 +157,16 @@ impl RFBState {
...
 
-    fn parse_request(&mut self, input: &[u8]) -> AppLayerResult {
+    fn parse_request(&mut self, flow: *const Flow, stream_slice: StreamSlice) -> AppLayerResult {
         // We're not interested in empty requests.
+        let input = stream_slice.as_slice();
+        let _pdu = Frame::new(
+            flow,
+            &stream_slice,
+            input,
+            input.len() as i64,
```
Comment (victorjulien) [inline rust/src/rfb/rfb.rs:167]: I think at this point in the parser we can't assume that input.len() == PDU length. input may contain a partial pdu, or more than one pdu.

**Comment by hsadia538** [comment]
> > I have written a suricat-verify test for the Pdu frame and it's working, Should I make a suricata-verify pr as well along with this?
> 
> Yes, please make it first and reference it in your suricata PR by adding a line in the message `suricata-verify-pr: NNN` where `NNN` is the SV PR number.

Updated the description with the pr reference

**Comment by victorjulien** [comment]
> > > I have written a suricat-verify test for the Pdu frame and it's working, Should I make a suricata-verify pr as well along with this?
> > 
> > 
> > Yes, please make it first and reference it in your suricata PR by adding a line in the message `suricata-verify-pr: NNN` where `NNN` is the SV PR number.
> 
> Updated the description with the pr reference

format is w/o the `#`

File: rust/src/rfb/rfb.rs (hunk 1 above)
Comment (hsadia538) [reply rust/src/rfb/rfb.rs:167]: Should I replace `-1_i64` here as it's done in the telnet parser?
In other parsers like `SIP` , `DNS` and `NFS`, I see` input.length()` being passed so I am assuming its ok for those parsers but not for `rfb`?

**Comment by hsadia538** [comment]
followed by #8322

**Pull request closed without merging**